internal/gatewayapi/clienttrafficpolicy.go

```diff
@@ -669,8 +670,31 @@ func translateHTTP1Settings(http1Settings *egv1a1.HTTP1Settings, httpIR *ir.HTTP
 					break
 				}
 			}
+			// second level of precedence - try to get a hostname from the HTTPRoutes
+			numMatchingRoutes := 0
 			if defaultHost == nil {
-				return fmt.Errorf("cannot set http10 default host on listener with only wildcard hostnames")
+				// When taken from the routes, a default hostname can only be chosen if there
+				// is exactly one HTTPRoute with a non-wildcard hostname configured.
+				for _, route := range httpIR.Routes {
+					if route.Hostname != "" && !strings.Contains(route.Hostname, "*") {
+						numMatchingRoutes++
+						// make the linter happy
+						theHost := route.Hostname
+						defaultHost = ptr.To(theHost)
+					}
+					if numMatchingRoutes > 1 {
+						defaultHost = nil
+						break
+					}
+				}
+			}
```

Comment:
the 2 if checks of `if defaultHost == nil {` did confuse me at first

can we delete L690-692 ?

so once we drop out of the `for` loop we can check for  `if numMatchingRoutes == 0` and `else if numMatchingRoutes != 1`

Reply:
Done.